R/forest_dot_plot.R

```diff
@@ -0,0 +1,173 @@
+# Function to extract legend from ggplot object
+#' @noRd
+extract_legend <- function(plot) {
+  g <- ggplotGrob(plot)
+  legends <- g$grobs[which(sapply(g$grobs, function(x) x$name) == "guide-box")]
+  legend <- legends[[1]]
+  return(legend)
+}
+
+#' Create a forest dot plot
+#'
+#' This function creates a composite forest dot plot using provided data and parameters.
+#'
+#' @param effects_table A data frame containing the effects data
+#' @param filters Character string specifying filters to apply. Default is "None"
+#' @param category Character string specifying the category. Default is "All"
+#' @param type_graph Character string specifying the type of graph. Default is "Absolute risk"
+#' @param type_risk Character string specifying the type of risk. Default is "Crude proportions"
+#' @param select_nnx Character string specifying whether to select NNX. Default is "Y"
+#' @param x_scale_fixed_free Character string specifying whether x-scale is fixed or free. Default is "Fixed"
+#' @param ci_method Character string specifying the CI method. Default is "Calculated"
+#' @param exclude_outcome Character string specifying an outcome to exclude. Default is "Liver"
+#'
+#' @return A ggplot object representing the composite forest dot plot
+#' @noRd
+forest_dot_plot <- function(effects_table,
+                            filters = "None",
+                            category = "All",
+                            type_graph = "Absolute risk",
+                            type_risk = "Crude proportions",
+                            select_nnx = "Y",
+                            x_scale_fixed_free = "Fixed",
+                            ci_method = "Calculated",
+                            exclude_outcome = "Liver") {
```

Comment:
The "Liver" outcome is specific to the example effects table. Should this be included in the examples rather than the function arguments?